pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -489,4 +491,18 @@ type (
 		// +optional
 		PublicKeyName *string `json:"publicKeyName,omitempty"`
 	}
+
+	// NodeGroupInstancesDistribution holds the configuration for spot instances
+	NodeGroupInstancesDistribution struct {
+		//+required
+		InstanceTypes []string `json:"instanceTypes,omitEmpty"`
+		// +optional
+		MaxPrice *float64 `json:"maxPrice,omitempty"`
+		//+optional
+		OnDemandBaseCapacity *int `json:"onDemandBaseCapacity,omitEmpty"`
+		//+optional
+		PercentageAboveBase *int `json:"percentageAboveBase,omitEmpty"`
```

Comment:
Why not use original term in ASG. like `OnDemandPercentageAboveBaseCapacity `

Reply:
I think we try to keep the names short in the API. In this case, I'm torn because it could be mistaken for the spot percentage. Let's see what @errordeveloper thinks.

Reply:
As far as I recall, we borrowed the short name from kops, but it would be a good idea to match the ASG API as this is 1-1 mapping.